doc/tutorials/image/image.txt

```diff
@@ -0,0 +1,210 @@
+@brief Tutorial for how to load and save images in mlpack.
+
+@page imagetutorial Image Utilities tutorial
+
+@section intro_imagetut Introduction
+
+Image dataset are becoming increasingly popular in deep learning.
+
+mlpack currently implements saving and loading of image. It uses [stb/](https://github.com/nothings/stb) to achieve this.
+
+@section toc_imagetut Table of Contents
+
+This tutorial is split into the following sections:
+
+ - \ref intro_imagetut
+ - \ref toc_imagetut
+ - \ref model_api_imagetut
+ - \ref imageinfo_api_imagetut
+ - \ref load_api_imagetut
+ - \ref save_api_imagetut
+
+@section model_api_imagetut Model API
+
+Image utilities supports loading and saving of image. Images can also be loaded from a directory if CXX version >= 17 as it requires <filesystem> header to navigate a directory.
+
+It supports filetypes "jpg", "png", "tga","bmp", "psd", "gif", "hdr", "pic", "pnm" for loading and "jpg", "png", "tga", "bmp", "hdr" for saving.
+
+The datatype associated is unsigned char to support RGB values in the range 1-255. To feed data into the network typecast of `arma::Mat` may be required. Images are stored in matrix as (width * height * channels, n_images). Therefore imageMatrix.col(0) would be first image if images are loaded in imageMatrix.
+
+@section imageinfo_api_imagetut ImageInfo
+
+ImageInfo class contains the metadata of the images.
+@code
+  /**
+   * Instantiate the ImageInfo object with the image width, height, channels.
+   *
+   * @param width Image width.
+   * @param height Image height.
+   * @param channels number of channels in the image.
+   */
+  ImageInfo(const size_t width,
+            const size_t height,
+            const size_t channels);
+@endcode
+Other public memebers include:
+  - flipVertical Flip the image vertical upon loading.
+  - quality Compression of the image if saved as jpg (0-100).
+
+@section load_api_imagetut Load
+
+Image can be loaded standalone or from a directory.
+
+Standalone loading of images.
+@code
+  /**
+   * Load the image file into the given matrix.
+   *
+   * @param filename Name of the image file.
+   * @param matrix Matrix to load the image into.
+   * @param info An object of ImageInfo class.
+   * @param fatal If an error should be reported as fatal (default false).
+   * @param transpose If true, transpose the matrix after loading.
+   * @return Boolean value indicating success or failure of load.
+   */
+   template<typename eT>
+   bool Load(const std::string& filename,
+             arma::Mat<eT>& matrix,
+             ImageInfo& info,
+             const bool fatal,
+             const bool transpose);
+@endcode
+
+Loading a test image. It also fills up the ImageInfo class object.
+@code
+data::ImageInfo info;
+data::Load("test_image.png", matrix, info, false);
+@endcode
+
+If the size of image is known beforehand, it can be passed into the constructor of ImageInfo to set the size of outputMatrix(the matrix into which the image is loaded).

```

Comment:
Might be a good idea to mention the fatal parameter here as well or just use the default value.

Reply:
Can we address the comment?